mountpoint-s3-client/src/s3_crt_client/put_object.rs

```diff
@@ -192,10 +217,15 @@ impl PutObjectRequest for S3PutObjectRequest {
         mut self,
         review_callback: impl FnOnce(UploadReview) -> bool + Send + 'static,
     ) -> ObjectClientResult<PutObjectResult, PutObjectError, Self::ClientError> {
+        if matches!(self.state, S3PutObjectRequestState::PendingWrite) {
```

Comment:
No need to wait in any case. We can just complete and report errors (if any). Added a comment.

Reply:
I see, in case of `open()` being followed by `flush()` (with no `write()`) in between, we'll instruct CRT to write an empty object to S3; we don't wait for headers in that case, but it seems not relevant for empty objects